### PR TITLE
Add sorting option to Template Library & Exporting to Elementor

### PR DESCRIPTION
**Pull request by HardeepAsrani**
### Summary
- Added sorting option which now allows you to change the order of the template by their date.
- Add "Save to Templates Cloud" to Elementor's context menu.
- Add Templates Library to Elementor.

Closes https://github.com/Codeinwp/templates-patterns-collection/issues/28

### CODE REVIEW

**Comment by irinelenache** [comment]
@HardeepAsrani found only a small problem:

- [ ] If i have a **sorting option activated** ( for example Name A-Z https://prnt.sc/wltara ) and i **delete one template** ( https://prnt.sc/wltbf0 ), the templates are sorted **back to default** ( Date new to old ) and the **sorting setting** still appears to be as the old one ( Name A-Z https://prnt.sc/wltcan ). This happens both on dashboard and elementor.

**Comment by rodica-andronache** [comment]
- [ ] similar to what Irinel mentioned, the `Search does not persist` when trying to also filter the results based on date, name of modified date
These are the results after searching "landing"
download (2)
These are the result after trying to filter the "landing" results by date
download (3)


- [ ] in RTL mode ( I've tested with the RTL Tested plugin ), the style doesn't load anymore
download (4)
compared to how it is in the current version
download (5)

**Comment by rodica-andronache** [comment]
- [ ] is there something needed to do to have the Page Templates and My Library listed here
download (6)
download (7)

In Elementor? I can't seem to understand how to test it, as I can't see any templates
download (8)
download (9)

**Comment by irinelenache** [comment]
@rodica-andronache 

> is there something needed to do to have the Page Templates and My Library listed here

The templates created and saved in elementor appear in My library section of elementor. About the Page templates section, i think the already existing templates are not elementor compatible and don't appear there.

**Comment by rodica-andronache** [comment]
@irinelenache thanks! is it working well for you? if so, can you please let me know what I'm missing? https://www.loom.com/share/342412a6301e45f5bedb200d3453f68b

**Comment by irinelenache** [comment]
@rodica-andronache You have to save the template directly in the elementor editor.
Edit the page as you want > Click on TPC button https://prnt.sc/wmn25r > Click on the save icon https://prnt.sc/wmn2di , give it a name and it will be saved 🚀

**Comment by rodica-andronache** [comment]
thanks, I got it now 👍 
then what I find confusing is that the Template cloud icon appears even when Neve Pro is not installed, and it looks like you can save the template and everything, even though you can't. And you only get this small notice in the bottom area 
download (3)

**Comment by HardeepAsrani** [comment]
Not any other issues are there, I suppose? @rodica-andronache

**Comment by selul** [comment]
@HardeepAsrani can you fix also the cypress tests

**Comment by HardeepAsrani** [comment]
Done.

**Comment by abaicus** [comment]
@HardeepAsrani I found two more things:

If you save a page as a template and reimport it inside the same page, or import it multiple times on another page, changing any value that uses CSS (like colors, backgrounds, fonts) on any of the iterations will change all of them.

If you change CSS values on an imported template and then import it again into the same page, the CSS is going to be reset back to the imported version for all the "templates" on the page.

I think this might be an issue with conflicting IDs (the imported elements keep their original ID).

---

Another thing that might be worth looking into is how the templates row item looks with elementor dark UI as it's not readable:

image

File: package.json [hunk 1]
```diff
@@ -37,25 +40,28 @@
...
   },
   "dependencies": {
-    "@wordpress/icons": "^2.7.0",
+    "@wordpress/icons": "^2.9.0",
     "classnames": "^2.2.6",
-    "fuse.js": "^6.4.1",
-    "query-string": "^6.13.6",
+    "fuse.js": "^6.4.6",
+    "i": "^0.3.6",
+    "npm": "^6.14.11",
```
Comment (abaicus) [inline package.json:61]: @HardeepAsrani I don't think these two should be here :see_no_evil:

Comment (HardeepAsrani) [reply package.json:61]: Yea, I was resetting somethings and npm added that. I'll remove it. :D

**Comment by abaicus** [comment]
I think we should also integrate the dashboard part with the new builders added to the Templates Cloud. 

i.e. we must make sure that the templates are publishable for Elementor too and that the user can select the builder inside the My Library & Page Templates views like in the figma mockup.

Also, we need to make sure that you can add individual templates to the demos, and that those will be importable.

https://www.figma.com/file/QklXyJh0XHVzq2h4IpPNaz/Neve-options?node-id=826%3A4584

To enable the publishing part of the plugin ping me on slack so I can guide you through the process.

**Comment by HardeepAsrani** [comment]
`Also, we need to make sure that you can add individual templates to the demos, and that those will be importable.`

Can you do that part? For the Publishing part, I can do it. I'll send you ping on Slack for instructions.

FYI, I've fixed the CSS & Dark Mode issue. Dark Mode in Modal works once you click inside if you change the mode again and again on the same page.

**Comment by HardeepAsrani** [comment]
Also added the importing for Elementor. Can you test the final changes and merge them? @abaicus

**Pull request merged**